sympy/matrices/expressions/matexpr.py

```diff
@@ -110,7 +110,11 @@ def __rmul__(self, other):
     def __pow__(self, other):
         if not self.is_square:
             raise ShapeError("Power of non-square matrix %s" % self)
-        if other is S.NegativeOne:
+        if self.is_ZeroMatrix:
+            return self
```

Comment:
It seems that  ``ZeroMatrix.__pow__`` already takes care of all powers of a ``ZeroMatrix``: zeroth, negative (could be ``if other < 0:``) and positive. Under which conditions would a ``ZeroMatrix`` appear as an argument of ``MatrixExpr.__pow__`` making this line necessary?